src/aics-image-viewer/components/App/ChannelUpdater.tsx

```diff
@@ -71,8 +71,14 @@ const ChannelUpdater: React.FC<ChannelUpdaterProps> = ({ index, view3d, image, v
       if (useControlPoints && controlPoints.length < 2) {
         return;
       }
+      const histogram = currentImage.getHistogram(index);
       const controlPointsToUse = useControlPoints ? controlPoints : rampToControlPoints(ramp);
-      const gradient = controlPointsToLut(controlPointsToUse);
+      // Convert control points from raw intensity values to histogram bin indices
+      const binIndexedControlPoints = controlPointsToUse.map((cp) => ({
+        ...cp,
+        x: histogram.findFractionalBinOfValue(cp.x),
+      }));
+      const gradient = controlPointsToLut(binIndexedControlPoints);
```

Comment:
am I thinking about this wrong, or should controlPointsToLut accept control points with raw intensity x coordinates?